src/AWSS3.jl

```diff
@@ -1075,6 +1083,31 @@ function s3_upload_part(
     return get_robust_case(Dict(response.headers), "ETag")
 end
 
+function s3_upload_part_copy(
+    aws::AbstractAWSConfig,
+    source,
+    upload,
+    part_number,
+    byte_range;
+    args=Dict{String,Any}(),
+    kwargs...,
+)
+    args["x-amz-copy-source-range"] = string(first(byte_range), '-', last(byte_range))
```

Comment:
```suggestion
    args["x-amz-copy-source-range"] = string("bytes=", first(byte_range), '-', last(byte_range))
```
https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html#API_UploadPartCopy_RequestSyntax

Reply:
```suggestion
    headers = Dict(
        "x-amz-copy-source-range" => string(
            "bytes=", first(byte_range), '-', last(byte_range)
        )
    )
    mergewith!(_merge, args, Dict("headers" => headers))
```
Otherwise it gets added as a query parameter rather than a header